platforms/chibios/drivers/i2c_master.c

```diff
@@ -111,11 +111,34 @@ i2c_status_t i2c_writeReg(uint8_t devaddr, uint8_t regaddr, const uint8_t* data,
     return chibios_to_qmk(&status);
 }
 
+i2c_status_t i2c_writeReg16(uint8_t devaddr, uint16_t regaddr, const uint8_t* data, uint16_t length, uint16_t timeout) {
+    i2c_address = devaddr;
+    i2cStart(&I2C_DRIVER, &i2cconfig);
+
+    uint8_t complete_packet[length + 2];
+    for (uint8_t i = 0; i < length; i++) {
```

Comment:
```suggestion
    for (uint16_t i = 0; i < length; i++) {
```
But the same bug is also in `i2c_writeReg()` above. And sending a large packet would also consume a large amount of stack space, which might also be problematic in some cases (the existing code probably does not do that though).